autogen/main/cluster.tf.tmpl

```diff
@@ -1094,6 +1109,21 @@ resource "google_container_node_pool" "windows_pools" {
         pod_pids_limit                         = lookup(each.value, "pod_pids_limit", null)
       }
     }
+
+    dynamic "sole_tenant_config" {
+      # node_affinity is currently the only member of sole_tenant_config
+      for_each = lookup(each.value, "node_affinity", null) != null ? [true] : []
+      content {
+        dynamic "node_affinity" {
+          for_each = lookup(each.value, "node_affinity", null) != null ? [lookup(each.value, "node_affinity", null)] : []
+          content {
+            key = jsondecode(node_affinity.value)["key"]
```

Comment:
Should we consider doing a lookup incase the provided JSON did not have the key?

Reply:
Yes! :)